synapse/handlers/auth.py

```diff
@@ -858,7 +860,11 @@ def _do_hash():
             return bcrypt.hashpw(password.encode('utf8') + self.hs.config.password_pepper,
                                  bcrypt.gensalt(self.bcrypt_rounds))
 
-        return make_deferred_yieldable(threads.deferToThread(_do_hash))
+        return make_deferred_yieldable(
+            threads.deferToThreadPool(
```

Comment:
why are we now using a threadpool here, ooi? (and what else does that threadpool do?)

Reply:
`deferToThread` implicitly uses the global reactor's threadpool -- so, this technically doesn't change behaviour, other than telling it explicitly what reactor (and what threadpool) we want to use. If we wished, we could add another threadpool explicitly for auth, here, rather than use the global one.